### PR TITLE
[FEATURE] Create user for first time google sigin automatically

### PR DESCRIPTION
**Pull request by ansoncht**

### CODE REVIEW

File: src/main/java/com/ocean/whale/service/user/UserService.java [hunk 1]
```diff
@@ -16,6 +19,13 @@ public UserService(FirestoreService firestoreService) {
         this.firestoreService = firestoreService;
     }
```
Comment (EdwinKam) [inline src/main/java/com/ocean/whale/service/user/UserService.java:21]: can yo rename `boolean getUserByUid` to something else? like isUserRegistered()

getUserByUid should be reserved for User getUserByUid() in the future.

File: src/main/java/com/ocean/whale/controller/AuthController.java [hunk 2]
```diff
@@ -27,8 +30,27 @@ public VerifyAuthResponse verifyAuth(@RequestParam String accessToken) {
         // http://localhost:8080/auth/verify?accessToken=eyJxxxx
         VerifyAuthResponse verifyAuthResponse = new VerifyAuthResponse();
         try {
-            Optional<String> uid = googleAuthService.getUid(accessToken);
-            verifyAuthResponse.setValidToken(uid.isPresent() && !uid.get().isBlank());
+            Optional<String> uidOpt = googleAuthService.getUid(accessToken);
+            if (uidOpt.isPresent()) {
+                String uid = uidOpt.get();
+                if (uid.isBlank()) {
```
Comment (EdwinKam) [inline src/main/java/com/ocean/whale/controller/AuthController.java:36]: can you simply the code by doing 
`if (uidOpt.isPresent() && !uidOpt.get().isBlank())`

File: src/main/java/com/ocean/whale/service/auth/GoogleAuthService.java [hunk 3]
```diff
@@ -16,4 +17,11 @@ public Optional<String> getUid(String accessToken) throws Exception {
         FirebaseToken decodedToken = FirebaseAuth.getInstance().verifyIdToken(accessToken);
```
Comment (EdwinKam) [inline src/main/java/com/ocean/whale/service/auth/GoogleAuthService.java:17]: or make this function return `Option.empty` if uid is `""`

File: src/main/java/com/ocean/whale/controller/AuthController.java (hunk 2 above)
Comment (ansoncht) [reply src/main/java/com/ocean/whale/controller/AuthController.java:36]: `uidOpt.isPresent()` and `uidOpt.get().isBlank()` mean two different scenarios. If we want more descriptive error, we should separate out. Combining them together requires us to make redundant call to `uidOpt.get()` as uid is needed to check and create user.

**Pull request merged**